geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/SplitArrays.scala

```diff
@@ -0,0 +1,19 @@
+/***********************************************************************
+* Copyright (c) 2013-2016 Commonwealth Computer Research, Inc.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Apache License, Version 2.0
+* which accompanies this distribution and is available at
+* http://www.opensource.org/licenses/apache2.0.php.
+*************************************************************************/
+
+package org.locationtech.geomesa.accumulo.index
+
+
+
+object SplitArrays {
+  val splitArraysMap: scala.collection.mutable.HashMap[Int, Seq[Array[Byte]]] =
```

Comment:
for thread safety, should make this a java ConcurrentHashMap and modify your getSplitArray to do a get/if-nulll-put. there is a slight chance that an array will be created more than once, but I believe it is faster than full synchronization.